xilem/src/view/transform.rs

```diff
@@ -1,43 +1,163 @@
+///
+/// The transform can be set using the methods on the return type.
+/// Transformations apply in order.
+/// That is, calling [`rotate`](Transformed::rotate) then [`translate`](Transformed::translate)
+/// will move the rotated widget.
+pub fn transformed<Child, State, Action>(child: Child) -> Transformed<Child, State, Action>
+where
+    Child: WidgetView<State, Action>,
+{
+    Transformed {
+        child,
+        transform: Affine::IDENTITY,
+        phantom: PhantomData,
+    }
+}
+
+/// The view for [`transformed`].
+pub struct Transformed<V, State, Action> {
+    child: V,
+    transform: Affine,
+    phantom: PhantomData<(State, Action)>,
+}
+
+impl<V, State, Action> Transformed<V, State, Action> {
     #[must_use]
-    fn rotate(mut self, radians: f64) -> Self {
-        let transform = self.transform_mut();
-        *transform = transform.then_rotate(radians);
+    /// Rotate the widget by `radians` radians about the origin of its natural location.
+    pub fn rotate(mut self, radians: f64) -> Self {
+        self.transform = self.transform.then_rotate(radians);
         self
     }
 
+    /// Scale the widget by `uniform` in each axis.
     #[must_use]
-    fn scale(mut self, uniform: f64) -> Self {
-        let transform = self.transform_mut();
-        *transform = transform.then_scale(uniform);
+    pub fn scale(mut self, uniform: f64) -> Self {
+        self.transform = self.transform.then_scale(uniform);
         self
     }
 
     #[must_use]
-    fn scale_non_uniform(mut self, x: f64, y: f64) -> Self {
-        let transform = self.transform_mut();
-        *transform = transform.then_scale_non_uniform(x, y);
+    /// Scale the widget by the given amount in each (2d) axis.
+    pub fn scale_non_uniform(mut self, x: f64, y: f64) -> Self {
+        self.transform = self.transform.then_scale_non_uniform(x, y);
         self
     }
 
     #[must_use]
-    fn translate(mut self, v: impl Into<crate::Vec2>) -> Self {
-        let transform = self.transform_mut();
-        *transform = transform.then_translate(v.into());
+    /// Displace the widget by `v` from its natural location.
+    pub fn translate(mut self, v: impl Into<crate::Vec2>) -> Self {
+        self.transform = self.transform.then_translate(v.into());
         self
     }
 
     #[must_use]
-    fn transform(mut self, v: impl Into<Affine>) -> Self {
-        *self.transform_mut() *= v.into();
+    /// Apply an arbitrary 2d transform to the widget.
+    pub fn transform(mut self, v: impl Into<Affine>) -> Self {
+        self.transform *= v.into();
         self
     }
 }
+
+mod private {
+    use crate::Affine;
+
+    /// The View state for the [Transformed](super::Transformed)
+    #[expect(
+        unnameable_types,
+        reason = "This type has no public API, and is only public due to trait visibility rules"
+    )]
+    pub struct TransformedState<ChildState> {
+        pub(super) child: ChildState,
+        pub(super) previous_transform: Affine,
```

Comment:
I don't think this extra attribute (and thus this whole state type) is really necessary?
Could just be a local variable in `View::rebuild`, couldn't it?

Reply:
Can you clarify how? Obviously in the case where an inner view makes a change, we need to rerun... But what about when the outer view changes?

Reply:
Nevermind, I checked this in more detail, it makes sense.